common/src/main/java/com/redhat/cloud/notifications/models/dto/v1/endpoint/EndpointDTO.java

```diff
@@ -0,0 +1,170 @@
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.CamelPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.EndpointPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.SystemSubscriptionPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.WebhookPropertiesDTO;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public final class EndpointDTO {
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private UUID id;
+
+    @JsonProperty(required = true)
+    @NotNull
+    @Size(max = 255)
+    private String name;
+
+    @JsonProperty(required = true)
+    @NotNull
+    private String description;
+
+    private Boolean enabled = Boolean.FALSE;
+
+    private EndpointStatusDTO status = EndpointStatusDTO.UNKNOWN;
+
+    @JsonProperty(required = true)
+    @Min(0)
+    private int serverErrors;
+
+    @JsonProperty(required = true)
+    @NotNull
+    private EndpointTypeDTO type;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(required = true)
+    @Size(max = 20)
+    private String subType;
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private LocalDateTime created;
+
+    @JsonProperty(required = true)
+    @JsonSubTypes({
+        @JsonSubTypes.Type(value = CamelPropertiesDTO.class, name = "camel"),
+        @JsonSubTypes.Type(value = CamelPropertiesDTO.class, name = "CAMEl"),
+        @JsonSubTypes.Type(value = SystemSubscriptionPropertiesDTO.class, name = "drawer"),
+        @JsonSubTypes.Type(value = SystemSubscriptionPropertiesDTO.class, name = "DRAWER"),
+        @JsonSubTypes.Type(value = SystemSubscriptionPropertiesDTO.class, name = "email_subscription"),
+        @JsonSubTypes.Type(value = SystemSubscriptionPropertiesDTO.class, name = "EMAIL_SUBSCRIPTION"),
+        @JsonSubTypes.Type(value = WebhookPropertiesDTO.class, name = "ansible"),
+        @JsonSubTypes.Type(value = WebhookPropertiesDTO.class, name = "ANSIBLE"),
+        @JsonSubTypes.Type(value = WebhookPropertiesDTO.class, name = "webhook"),
+        @JsonSubTypes.Type(value = WebhookPropertiesDTO.class, name = "WEBHOOK")
```

Comment:
Why do we need both lowercase and uppercase names?

Reply:
Because even though we had the following code...

https://github.com/RedHatInsights/notifications-backend/blob/307c05470b75750baef111d41cf07d7de789ee2c/backend/src/main/java/com/redhat/cloud/notifications/routers/RegisterCustomModuleCustomizer.java#L9-L14

... I did not understand why we were having test failures with "the provided type is not valid". The thing is that the tests still use the internal model, and when I removed the `@JsonProperty` definitions that renamed the enum values to lowercase, the endpoints under test were complaining because they were expecting the types in lowercase.

I decided to use the leave the internal entities in the tests to see if using the DTOs in the endpoints under test would break anything.